src/main/java/com/myhome/services/springdatajpa/CommunitySDJpaService.java

```diff
@@ -113,29 +118,45 @@ public Community createCommunity(CommunityDto communityDto) {
   @Override
   public Set<String> addHousesToCommunity(String communityId, Set<CommunityHouse> houses) {
     Optional<Community> communitySearch = communityRepository.findByCommunityId(communityId);
-    Set<String> addedHousesIds = communitySearch.map(community -> {
-      Set<CommunityHouse> savedHouses = houses
-          .stream()
-          .filter(house -> communityHouseRepository.findByHouseId(house.getHouseId()) != null)
-          .map(house -> house.withCommunity(community))
-          .filter(house -> community.getHouses().add(house))
-          .collect(Collectors.toSet());
-      communityHouseRepository.saveAll(savedHouses);
-      Set<String> housesIds = communityRepository.save(community).getHouses()
-          .stream()
-          .map(house -> house.getHouseId())
-          .collect(Collectors.toSet());
-      return housesIds;
+
+    return communitySearch.map(community -> {
+      Set<String> addedIds = new HashSet<>();
+
+      houses.forEach(house -> {
+        house = communityHouseRepository.findByHouseId(house.getHouseId());
+
+        if (house != null) {
+          Community houseCommunity = house.getCommunity();
+
+          if (houseCommunity != null) {
+            CommunityHouse finalHouse = house;
+            communityRepository.findByCommunityId(houseCommunity.getCommunityId())
+              .map(community1 -> {
+                community1.getHouses().remove(finalHouse);
```

Comment:
Why we delete here a house? I don't get this logic

Reply:
Because if you don't remove it, when adding a house to a community it would cause a SQL error if that house was added to a different community before since it would violate a unique key constraint so you need to remove it from the old community before adding it to the new one

Reply:
@jmprathab Can you explain that logic? I mean it is strange to delete house from one community and add it to another in addHouse method.

Reply:
A house can't be in 2 communities at once so that's why it needs to be removed from another if it was already in a community before it wants to be added to the new community or else it means you'll violate the unique key for house id trying to add it to a second community.

But I suppose we could remove logic removing the house in the add house method and call a method that removes the house from the old community

Reply:
Yes, I understand the code problem but the domain problem seems to be strange.

How we can know that such person can 'disconnect' and 'connect' house between communities?
Maybe I miss something.

Reply:
I get what you mean. Wouldn't imagine somebody can move a house from community to community! I just found there's no code to add a new house to a new community. All the houses in the system at runtime have already been loaded in and they all seem to have communities already when retrieved from the repository hence why I added that code. So I'm not sure how to resolve this. As when adding a house to a community, the code checks if that house exists first before adding it, so it had to be added before hand before it can be added to a community! So you can't create a new house here. @mprathab what are your thoughts?

Reply:
> @jmprathab Can you explain that logic? I mean it is strange to delete house from one community and add it to another in addHouse method.

Yeah, this is not the business use case. We should never let a house to be moved/added to another community.

Reply:
> I get what you mean. Wouldn't imagine somebody can move a house from community to community! I just found there's no code to add a new house to a new community. All the houses in the system at runtime have already been loaded in and they all seem to have communities already when retrieved from the repository hence why I added that code. So I'm not sure how to resolve this. As when adding a house to a community, the code checks if that house exists first before adding it, so it had to be added before hand before it can be added to a community! So you can't create a new house here. @jmprathab what are your thoughts?

I believe the implementation has to be something like this.
1. Do a single pass to check if the houses we are going to add is already part of the community(Check for name uniqueness). If a house with the same name exists, we should just return an error from the API without proceeding.
2. Persist all the community houses.
3. Update community to add all the newly persisted house and persist community.

Reply:
So, this API needs a bit of refactoring. We should never get House Id from the user to add it. The only data this API needs to take is a list of house names.

Reply:
> > I get what you mean. Wouldn't imagine somebody can move a house from community to community! I just found there's no code to add a new house to a new community. All the houses in the system at runtime have already been loaded in and they all seem to have communities already when retrieved from the repository hence why I added that code. So I'm not sure how to resolve this. As when adding a house to a community, the code checks if that house exists first before adding it, so it had to be added before hand before it can be added to a community! So you can't create a new house here. @jmprathab what are your thoughts?
> 
> I believe the implementation has to be something like this.
> 
> 1. Do a single pass to check if the houses we are going to add is already part of the community(Check for name uniqueness). If a house with the same name exists, we should just return an error from the API without proceeding.
> 2. Persist all the community houses.
> 3. Update community entity to add all the newly persisted house and persist community.
> 
> So, this API needs a bit of refactoring. We should never get House Id from the user to add it. The only data this API needs to take is a list of house names.

That's right! :-)

Reply:
> > @jmprathab Can you explain that logic? I mean it is strange to delete house from one community and add it to another in addHouse method.
> 
> Yeah, this is not the business use case. We should never let a house to be moved/added to another community. Having said that, there can be two houses in two different communities with the same name and name of the house within a community has to be unique.

The issue with this though is that you will get that SQL Unique Key violation error. We'll have to see if there's another way to fix that

Reply:
> > I get what you mean. Wouldn't imagine somebody can move a house from community to community! I just found there's no code to add a new house to a new community. All the houses in the system at runtime have already been loaded in and they all seem to have communities already when retrieved from the repository hence why I added that code. So I'm not sure how to resolve this. As when adding a house to a community, the code checks if that house exists first before adding it, so it had to be added before hand before it can be added to a community! So you can't create a new house here. @jmprathab what are your thoughts?
> 
> I believe the implementation has to be something like this.
> 
>     1. Do a single pass to check if the houses we are going to add is already part of the community(Check for name uniqueness). If a house with the same name exists, we should just return an error from the API without proceeding.
> 
>     2. Persist all the community houses.
> 
>     3. Update community entity to add all the newly persisted house and persist community.
> 
> 
> So, this API needs a bit of refactoring. We should never get House Id from the user to add it. The only data this API needs to take is a list of house names.

So when you take this list of house names @jmprathab, it's supposed to create a new house entity with a new random id? Or how do you want to approach it?

Reply:
> So when you take this list of house names @jmprathab, it's supposed to create a new house entity with a new random id? Or how do you want to approach it?

@edwardUL99 That's right. It should create a new `CommunityHouse` with a random UUID and add it to the `Community`